lib/proxy/http.js

```diff
@@ -177,7 +178,8 @@ function ShinyProxy(router, schedulerRegistry) {
         req.headers['shiny-shared-secret'] = appWorkerHandle.endpoint.getSharedSecret();
         // Remember that we will null out req and res in the proxy, so be careful
         // trying to subscribe to events on them after we've started this proxy.
-        appWorkerHandle.proxy.proxyRequest(req, res, buffer);
+        appWorkerHandle.proxy.web(req, res);
+        req.resume();
       })
       .fail(function(err) {
         if (acquired){
```

Comment:
And line 97, as the internal router would have handled the request and we should unblock the proxy.